### PR TITLE
Add test for rust-lang#114912

### LINKED ISSUE #114912: debug ice: attempted to add with overflow
### Code
`rustc --edition=2021`
```Rust
async fn main() {
    [0usize; 0xffff_ffff_ffff_ffff].await;
}

```

### Meta

`rustc --version --verbose`:
```
c94cb834d05888ef5beeae2cc460156f435e2fb3
```

### Error output

```
error[E0277]: `[usize; usize::MAX]` is not a future
 --> D01319EFC4D41E1760BDE6D24D4EAEBDED06776A984903041C4E462ABADDEE9A.rs:2:37
  |
2 |     [0usize; 0xffff_ffff_ffff_ffff].await;
  |                                    -^^^^^
  |                                    ||
  |                                    |`[usize; usize::MAX]` is not a future
  |                                    help: remove the `.await`
  |
  = help: the trait `Future` is not implemented for `[usize; usize::MAX]`
  = note: [usize; usize::MAX] must be a future or must implement `IntoFuture` to be awaited
  = note: required for `[usize; usize::MAX]` to implement `IntoFuture`
```

### PR DESCRIPTION
**Pull request by apekros**
Closes #114912

### CODE REVIEW

File: tests/ui/async-await/debug-ice-attempted-to-add-with-overflow.rs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// check-fail
+// edition:2021
+
+// test for issue-114912 - debug ice: attempted to add with overflow
+
+async fn main() {
```
Comment (vincenzopalazzo) [inline tests/ui/async-await/debug-ice-attempted-to-add-with-overflow.rs:6]:
```suggestion
fn main() {
```

We do not support async main so we can remote it? of there is any reason that I am missing to add it?

File: tests/ui/async-await/debug-ice-attempted-to-add-with-overflow.rs [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+    //~^ ERROR E0752
+    [0usize; 0xffff_ffff_ffff_ffff].await;
+    //~^ ERROR E0277
```
Comment (vincenzopalazzo) [inline tests/ui/async-await/debug-ice-attempted-to-add-with-overflow.rs:9]:
```suggestion
    //~^ error[E0277]: `[usize; usize::MAX]` is not a future
```

As adding the error code is super helpful usually I do not know what it means,
and I preferer to read the error. This also prevents the message content regression.

**Review by vincenzopalazzo — COMMENTED** [review]
I do no know if I am missing some context here :) so feel free to correct my review if needed 😸

File: tests/ui/async-await/debug-ice-attempted-to-add-with-overflow.rs (hunk 1 above)
Comment (vincenzopalazzo) [reply tests/ui/async-await/debug-ice-attempted-to-add-with-overflow.rs:6]: Ah I see, ok it is fine for me.

another way to write the same thing is 

```rust 
fn main() {
   let _ = async { 
      [0usize; 0xffff_ffff_ffff_ffff].await;
   };
}
```

In this way, you have no longer the problem of the `async fn main`

Comment (apekros) [reply tests/ui/async-await/debug-ice-attempted-to-add-with-overflow.rs:6]: Would you prefer it this way, or okay with how it is now? Thanks for the tip :)

Comment (vincenzopalazzo) [reply tests/ui/async-await/debug-ice-attempted-to-add-with-overflow.rs:6]: >Would you prefer it this way, or okay with how it is now? Thanks for the tip :)

for me, it is fine either way, with the second way you avoid just to generate a second error

error[E0752]: `main` function is not allowed to be `async`

but maybe a t-compiler member will have some preference, for me there is no problem :)

**Review by vincenzopalazzo — APPROVED** [review]

File: tests/ui/async-await/debug-ice-attempted-to-add-with-overflow.rs (hunk 1 above)
Comment (apekros) [reply tests/ui/async-await/debug-ice-attempted-to-add-with-overflow.rs:6]: Awesome, thanks mate.

**Pull request merged**